evergreen/src/e/edit/WorkspaceFileList.java

```diff
@@ -181,11 +259,34 @@ private void fireListeners(final boolean isNowValid) {
         synchronized (listeners) {
             for (final Listener l : listeners) {
                 // Ensure we're running on the EDT.
-                EventQueue.invokeLater(new Runnable() {
-                    public void run() {
-                        l.fileListStateChanged(isNowValid);
-                    }
-                });
+                GuiUtilities.invokeLater(() -> { l.fileListStateChanged(isNowValid); });
+            }
+        }
+    }
+    
+    private void fireFileCreated(final String filename) {
```

Comment:
this and the following 2 methods maybe deserve a helper method like this:
```
private void notifyListeners(action:Consumer<Listener>) {
     for (final Listener l : listeners) {
        GuiUtilities.invokeLater(() -> action(l));
    }
}
```
which then can be called with
```
notifyListeners(l ->  l.fileCreated(filename));
```